modules/services/utils/src/main/java/au/com/shiftyjelly/pocketcasts/utils/extensions/String.kt

```diff
@@ -65,6 +66,12 @@ fun String.removeNewLines(): String {
     return this.replace("[\n\r]".toRegex(), "")
 }
 
+fun String.removeAccents() =
+    Normalizer.normalize(this, Normalizer.Form.NFD)
+        .replace("\\p{Mn}+".toRegex(), "")
+        .replace("\u0141", "L") // Remove L with stroke
+        .replace("\u0142", "l") // Remove l with stroke
```

Comment:
I'm afraid that if we need to handle `Ł` and `ł` manually that it might not work for more diacritics. Could you writ a test for this function that checks more of them?

The issue might be with `java.text.Normalizer` and we might need to use `android.icu.text.Normalizer2` on SDK > 24 and fallback to `j.t.Normalizer` on SDK 23. I think an edge case for SDK 23 would be an ok compromise if the issue is with `j.t.Normalizer`.

Reply:
`android.icu.text.Normalizer2` also has the same problem. From what I've found, most normalizers do not handle `Ł` and `ł` properly. Even [stripAccents](https://commons.apache.org/proper/commons-lang/javadocs/api-release/src-html/org/apache/commons/lang3/StringUtils.html#line.1382) provided by `org.apache.commons.lang3.StringUtils` replaces "L with stroke" (Ł [\u0141], ł[\u0142]).

Reply:
Ah, got it! Thanks for the explanation. Could you add the tests though? I think it would help to protect this implementation from any accidental changes.

Reply:
I covered some more diacritics in 82c9527.